src/badfish/main.py

```diff
@@ -1353,115 +1223,175 @@ async def toggle_boot_device(self, device):
+                    for m in data["Members"]:
+                        vm_path["members"].append(m.get("@odata.id"))
+            except (ValueError, KeyError):
+                raise BadfishException("Not able to access virtual media config.")
         else:
-            raise BadfishException("No VirtualMedia endpoint found")
-
-        return vms
+            try:
+                vm_path = []
+                for m in data["Members"]:
+                    vm_path.append(m.get("@odata.id"))
+            except (ValueError, KeyError):
+                raise BadfishException("Not able to access virtual media config.")
+        return vm_path
 
     async def check_virtual_media(self):
-        vms = await self.get_virtual_media()
-        for vm in vms:
-            disc_url = "%s%s" % (self.host_uri, vm)
-            disc_response = await self.get_request(disc_url)
+        vm_config = await self.get_virtual_media_config()
+        if self.vendor == "Supermicro":
+            if vm_config.get("count") == 0:
+                self.logger.info("No virtual media mounted.")
+                return False
+            else:
+                vm_config = vm_config["members"]
+
+        inserted = False
+        for vm in vm_config:
+            _uri = "%s%s" % (self.host_uri, vm)
+            _response = await self.get_request(_uri)
             try:
-                raw = await disc_response.text("utf-8", "ignore")
-                disc_data = json.loads(raw.strip())
-                self.logger.info(f"{disc_data.get('Id')}:")
-                self.logger.info(f"    Name: {disc_data.get('Name')}")
-                self.logger.info(f"    ImageName: {disc_data.get('ImageName')}")
-                self.logger.info(f"    Inserted: {disc_data.get('Inserted')}")
+                raw = await _response.text("utf-8", "ignore")
+                _data = json.loads(raw.strip())
+                self.logger.info(f"{_data.get('Id')}:")
+                self.logger.info(f"    Name: {_data.get('Name')}")
+                self.logger.info(f"    ImageName: {_data.get('ImageName')}")
+                self.logger.info(f"    Inserted: {_data.get('Inserted')}")
+                if str(_data.get("Inserted")).lower() == "true" and "CD" in str(_data.get("Id")):
+                    inserted = True
             except ValueError:
-                raise BadfishException(
-                    "There was something wrong getting values for VirtualMedia"
-                )
-
+                raise BadfishException("There was something wrong getting values for VirtualMedia")
+        return inserted
+
+    async def mount_virtual_media(self, path):
+        vm_config = await self.get_virtual_media_config()
+        _headers = {"Content-Type": "application/json"}
+        if self.vendor == "Supermicro":
+            split_path = str(path).split("/", 3)
+            _payload = {"Host": "/".join(split_path[:3]), "Path": "/" + split_path[-1], "Username": "", "Password": ""}
```

Comment:
Afraid this code might be a little error prone with some corner case URLs. Safer to use `urllib.parse.urlparse`

``` bash
>>> from urllib.parse import urlparse
>>> urlparse(path)
ParseResult(scheme='http', netloc='storage.example.com', path='/folder/linux.iso', params='', query='', fragment='')
>>> urlparse(path).netloc
'storage.example.com'
>>> urlparse(path).path
'/folder/linux.iso'
```

Reply:
Fixed with [this commit](https://github.com/redhat-performance/badfish/pull/324/commits/e593766fc05c12e635ac5dd70c306df7e3c84a22).